### PR TITLE
feat: add support for macOS packages

### PR DESCRIPTION
**Pull request by amdprophet**
Adds support for amd64 & arm64 macOS packages.

The `TARGET` environment variable is now necessary to build this project. It must be the name of a file inside of the `targets` directory minus the extension (e.g. `TARGET=otc_darwin_amd64_productbuild`). Multiple packages can no longer be built simultaneously as it complicated the code and CMake/CPack aren't meant to be used that way.

I couldn't find a way to generate the uninstall script using CMake. As a result we will have to ensure to update the script when any new files or directories are added to the package. As an attempt to prevent the case where we forget to update the script, I've added a `test_package` CI job that verifies that no files or directories remain after uninstallation.

Package signing is not included in this PR but can be implemented separately.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -34,18 +38,24 @@ project(otelcol-sumo
...
 
-# Include package functions
-include("${PACKAGES_DIR}/otc_linux_amd64_deb.cmake")
-include("${PACKAGES_DIR}/otc_linux_amd64_rpm.cmake")
+set(TARGET "$ENV{TARGET}")
+
+if("${TARGET}" STREQUAL "")
+  message(WARNING "TARGET variable was not set, skipping build...")
+  return()
+endif()
```
Comment (sumo-drosiek) [inline CMakeLists.txt:46]: Could we list here supported targets as well?

File: assets/productbuild/uninstall.sh
Comment (sumo-drosiek) [inline assets/productbuild/uninstall.sh:1]: Could we validate it with shellcheck?
Also I believe we should add shellcheck to github actions like in other bash containing repositories

File: components/otelcol-sumo.cmake [hunk 2]
```diff
@@ -8,11 +9,44 @@ macro(default_otc_linux_install)
...
+  install_otc_sumologic_yaml()
+  install_otc_darwin_hostmetrics_yaml()
+  install_otc_binary()
+  install_otc_uninstall_script()
+endmacro()
+
+macro(create_otc_components)
+  cpack_add_component_group("otelcol-sumo-group"
+    DISPLAY_NAME "OpenTelemetry Collector"
+    #DESCRIPTION "example group description"
```
Comment (sumo-drosiek) [inline components/otelcol-sumo.cmake:27]: We should remove it before merge, as well as other unneeded comments

Comment (amdprophet) [reply components/otelcol-sumo.cmake]: I think I got it all other than one block that I'd like to keep as it should be uncommented once CPack adds a way to configure `hostArchitectures` for macOS packages.

**Review by sumo-drosiek — APPROVED** [review]

**Pull request merged**